var/spack/repos/builtin/packages/yaml-cpp/package.py

```diff
@@ -39,7 +40,14 @@ class YamlCpp(CMakePackage):
     variant('pic',   default=True,
             description='Build with position independent code')
 
-    depends_on('boost', when='@:0.5.3')
+    depends_on('boost@:1.66.99', when='@:0.5.3')
+
+    conflicts('%gcc@:4.8', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
+    conflicts('%clang@:3.3.0', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
+    conflicts('%clang@:4.0.0-apple', when='@0.6.0:', msg="versions 0.6.0: require c++11 support")
```

Comment:
This doesn't actually work the way you would expect:
```console
$ spack python
Spack version 0.11.2
Python 3.6.4, Darwin x86_64
>>> from spack.version import Version
>>> Version('4.0.0') < Version('4.0.0-apple')
True
```
So this would incorrectly filter out some versions of non-Apple clang.

@tgamblin Any way to solve this? The simplest thing I can think of is to create an entirely separate compiler for `%apple-clang`, but that seems like overkill. It would be even nicer if we could directly query the compiler and say something like:
```python
requires('c++11')
```
instead of enumerating through every compiler and duplicating the work in multiple packages. This feature was also requested by @svenevs in #7672.